### ISSUE TITLE
Debugger hangs on Validating Breakpoint Location, messagebox cannot be dismissed

### ISSUE THREAD

**Issue by abelbraaksma** [dotnet/fsharp#9216]
The following messagebox is shown during a debugging session in F#, VS 2019 IDE. When clicking Cancel, it reappears, or disappears entirely, surviving as a hidden window, but still blocking any UI input to the VS IDE.

The messagebox has the parent HWND set to a window called "Hidden Window", which in turn has no parent HWND (`NULL`), seen with Spy++. Meaning you can Alt-Tab away from it, or have it obscure other windows not related to VS. This part seems to be a VS IDE specific bug.

The messagebox before it disappeared:

![Validating Breakpoint Location screenshot](https://user-images.githubusercontent.com/16015770/82152460-96f35300-9861-11ea-9733-b4ddf2f1f4d1.JPG)

The messagebox Window, not under the IDE:

![image](https://user-images.githubusercontent.com/16015770/82152451-8216bf80-9861-11ea-893d-3473e08e14cd.png)

A trace is here: https://developercommunity.visualstudio.com/content/problem/1035404/debugger-hangs-on-validating-breakpoint-location-m.html (trace available to MS employees only).

**Repro steps**

Unclear: this is the first time I consciously see this. What I did was "normal" debugging, and just tried to set a new breakpoint in a formerly valid breakpoint location.

I did notice, during debugging, that there were no tooltips of any variable. They did, however, appear in the Locals window. This suggests the PDB is wrong/corrupt/faulty somehow.

**Expected behavior**

Hitting F9 should set a new breakpoint. If it can't, it should not block the whole IDE.

**Actual behavior**

Messagebox appears, dismissing it makes it reappear, though the reappearance can be behind the main window, or hidden completely. Either way, it blocks any UI input (mouse appears either as a wait-state icon, or a normal pointer, but you cannot click anything).

**Known workarounds**

None that I know of. I will, of course, try the same session again, to find out whether it is a fluke or something more.

**Related information**

Recent non-preview version of VS 2019 IDE, Community Edition.

A trace is [available here](https://developercommunity.visualstudio.com/content/problem/1035404/debugger-hangs-on-validating-breakpoint-location-m.html).

**Comment by brettfo** [comment]
Internal tracking bug: [1126487](https://devdiv.visualstudio.com/DevDiv/_queries/edit/1126487/?triage=true)

**Comment by dsyme** [comment]
I believe I've seen this too.  Tricky to make progress without a repro.  I'll leave it open - it would be great to dig out the problem.

**Comment by TonyHenrique** [comment]
Have just git this issue while editing this code:
If I go to and place a **breakpoint** on line 20 then put the cursor before the **when** and press **Enter** then that Message Box appears.
```
// Learn more about F# at http://docs.microsoft.com/dotnet/fsharp

open System

/// Decodes a Roman Number string to a int number // https://twitter.com/nikoloz_p/status/1421218836896960515
let DecodeRomanNumeral RomanNumeral =
    let DigitToNumber = function
    | 'I' -> 1
    | 'V' -> 5
    | 'X' -> 10
    | 'L' -> 50
    | 'C' -> 100
    | 'D' -> 500
    | 'M' -> 1000
    | c -> failwith (sprintf "Invalid digit %c" c)

    let rec sum' Numbers = 
        match Numbers with
        | [] ->0
        | x::y::rest when x < y -> y - x + sum' rest
        | x::rest -> x + sum' rest
    
    RomanNumeral 
        |> Seq.map DigitToNumber
        |> Seq.toList
        |> sum'


[<EntryPoint>]
let main argv =
    let message = DecodeRomanNumeral "XVXV"  
    printfn "Hello world %i" message
    0

```

![image](https://user-images.githubusercontent.com/6307623/127739749-72824cad-d295-4704-9dfc-98c3354df008.png)

**Comment by dsyme** [comment]
@TonyHenrique I don't repro this - could you list out exact versions you're using?  Thanks

**Comment by TonyHenrique** [comment]
Below are the versions I was using when I saw that issue.
I also have recorded it and submitted on Visual Studio Feedback:

https://developercommunity.visualstudio.com/t/Validating-breakpoint-location-when-e/1491629?port=1025&fsid=8f683d68-4da8-4d00-b527-c8e74e55005e&entry=myfeedback&ref=native&refTime=1628259021623&refUserId=72063368-4a63-4d0d-a595-6d21acf4cd4e

```
Microsoft Visual Studio Community 2022 Preview
Version 17.0.0 Preview 2.1
VisualStudio.17.Preview/17.0.0-pre.2.1+31521.260
Microsoft .NET Framework
Version 4.8.04084

Installed Version: Community

.NET Core Debugging with WSL 2   1.0
.NET Core Debugging with WSL 2

ASP.NET and Web Tools 2019   17.0.303.37752
ASP.NET and Web Tools 2019

ASP.NET Web Frameworks and Tools 2019   17.0.303.37752
For additional information, visit https://www.asp.net/

Azure App Service Tools v3.0.0   17.0.303.37752
Azure App Service Tools v3.0.0

Azure Functions and Web Jobs Tools   17.0.303.37752
Azure Functions and Web Jobs Tools

C# Tools   4.0.0-2.21362.3+5f7ddf6f02d40036fdbf823cd9910027da277acd
C# components used in the IDE. Depending on your project type and settings, a different version of the compiler may be used.

Common Azure Tools   1.10
Provides common services for use by Azure Mobile Services and Microsoft Azure Tools.

Microsoft JVM Debugger   1.0
Provides support for connecting the Visual Studio debugger to JDWP compatible Java Virtual Machines

Microsoft Library Manager   2.1.134+45632ee938.RR
Install client-side libraries easily to any web project

Microsoft MI-Based Debugger   1.0
Provides support for connecting Visual Studio to MI compatible debuggers

Microsoft Visual Studio Tools for Containers   1.2
Develop, run, validate your ASP.NET Core applications in the target environment. F5 your application directly into a container with debugging, or CTRL + F5 to edit & refresh your app without having to rebuild the container.

NuGet Package Manager   6.0.0
NuGet Package Manager in Visual Studio. For more information about NuGet, visit https://docs.nuget.org/

ProjectServicesPackage Extension   1.0
ProjectServicesPackage Visual Studio Extension Detailed Info

Razor (ASP.NET Core)   17.0.0.2135615+c94df2e3bebc72c937da0389b22d730b5ea7b19b
Provides languages services for ASP.NET Core Razor.

SQL Server Data Tools   17.0.62107.02200
Microsoft SQL Server Data Tools

TypeScript Tools   17.0.0701.2001
TypeScript Tools for Microsoft Visual Studio

Visual Basic Tools   4.0.0-2.21362.3+5f7ddf6f02d40036fdbf823cd9910027da277acd
Visual Basic components used in the IDE. Depending on your project type and settings, a different version of the compiler may be used.

Visual F# Tools   17.0.0-beta.21352.5+3af67cd8ffd73b2dc443e1e24dc0cf28f7e1c608
Microsoft Visual F# Tools

Visual Studio Code Debug Adapter Host Package   1.0
Interop layer for hosting Visual Studio Code debug adapters in Visual Studio

Visual Studio Container Tools Extensions   1.0
View, manage, and diagnose containers within Visual Studio.

Visual Studio IntelliCode   2.2
AI-assisted development for Visual Studio.

Visual Studio Tools for Containers   1.0
Visual Studio Tools for Containers
```

**Comment by dsyme** [comment]
@TonyHenrique Hmmm... Strange, I've checked the code for validating `when` breakpoints and can't see a problem.  Had you launched debugging?

**Comment by TonyHenrique** [comment]
Yes, I used Start Debugging - F5 .
The message appears while not debugging. When the breakpoint is ON, and I am editing and press enter before the **where** keyword.
Right now when I opened to check it again, Visual Studio hangs on this:
![image](https://user-images.githubusercontent.com/6307623/128525181-bbbe838f-b6fa-4769-b837-db4e91dac247.png)

I had to close Visual Studio
reopened the project
_An error was encountered while opening associated documents the last time this solution was loaded. Document load is being skipped during this solution load in order to avoid that error._

![image](https://user-images.githubusercontent.com/6307623/128525945-aec26907-c38d-458d-bed4-e59e7e932059.png)
